ibis/backends/flink/tests/test_time_travel.py

```diff
@@ -0,0 +1,195 @@
+from __future__ import annotations
+
+import os
+import tempfile
+from pathlib import Path
+
+import pandas as pd
+import pandas.testing as tm
+import pyarrow as pa
+import pytest
+
+import ibis
+import ibis.common.exceptions as exc
+import ibis.expr.datatypes as dt
+import ibis.expr.schema as sch
+from ibis.backends.conftest import TEST_TABLES
+from ibis.backends.flink.tests.utils import download_jar_for_package, get_catalogs
+from ibis.backends.tests.errors import Py4JJavaError
+
+
+@pytest.fixture
+def use_hive_catalog(con):
+    # Flink related
+    download_jar_for_package(
```

Comment:
Can you do this in the flink dockerfile?

Reply:
Adding JAR files in the container in this case won't help because the Flink specific tests do not run on the dockerized env, but on the local env as [described above](https://github.com/ibis-project/ibis/pull/8517#discussion_r1509780167).

Reply:
Downloading JAR's is failing on CI:
```
package_name = 'apache-flink'
jar_name = 'paimon-flink-1.18-0.8-20240301.002155-30'
jar_url = 'https://repository.apache.org/content/groups/snapshots/org/apache/paimon/paimon-flink-1.18/0.8-SNAPSHOT/paimon-flink-1.18-0.8-20240301.002155-30.jar'

    def download_jar_for_package(
        package_name: str,
        jar_name: str,
        jar_url: str,
    ):
        import os
        from importlib import metadata
    
        import requests
    
        # Find the path to package lib
        try:
            distribution = metadata.distribution(package_name)
            lib_path = distribution.locate_file("")
        except metadata.PackageNotFoundError:
            lib_path = None
    
        # Check if the JAR already exists
        jar_path = os.path.join(lib_path, "pyflink/lib", f"{jar_name}.jar")
        if os.path.exists(jar_path):
            return jar_path
    
        # Download the JAR
        response = requests.get(jar_url, stream=True)
        if response.status_code != 200:
>           raise SystemError(
                f"Failed to download the JAR file \n"
                f"\t jar_url= {jar_url} \n"
                f"\t response.status_code= {response.status_code}"
            )
E           SystemError: Failed to download the JAR file 
E           	 jar_url= https://repository.apache.org/content/groups/snapshots/org/apache/paimon/paimon-flink-1.18/0.8-SNAPSHOT/paimon-flink-1.18-0.8-20240301.002155-30.jar 
E           	 response.status_code= 404
```

The "easy" way to deal with this whole thing of downloading JAR's in Python would be skipping these tests by default and leaving them only for manual local testing.